src/triage/component/catwalk/model_trainers.py

```diff
@@ -267,17 +282,29 @@ def _train_and_store_model(
 
         unique_parameters = self.unique_parameters(parameters)
 
-        model_group_id = self.model_grouper.get_model_group_id(
-            class_path, unique_parameters, matrix_store.metadata, self.db_engine
-        )
+               
+        if model_hash is None and retrain and model_group_id:
```

Comment:
I think I might not be following the flow here entirely, but shouldn't the `model_hash` always be calculated upstream of the call to `_train_and_store_model()`?